ddtrace/tracer/log.go

```diff
@@ -0,0 +1,107 @@
+	SampleRate            string            `json:"sample_rate"`             // The default sampling rate for the rules sampler
+	SamplingRules         []SamplingRule    `json:"sampling_rules"`          // Rules used by the rules sampler
+	SamplingRulesError    string            `json:"sampling_rules_error"`    // Any errors that occurred while parsing sampling rules
+	Tags                  map[string]string `json:"tags"`                    // Global tags
+	RuntimeMetricsEnabled bool              `json:"runtime_metrics_enabled"` // Whether or not runtime metrics are enabled
+	HealthMetricsEnabled  bool              `json:"health_metrics_enabled"`  // Whether or not health metrics are enabled
+	ApplicationVersion    string            `json:"dd_version"`              // Version of the user's application
+	Architecture          string            `json:"architecture"`            // Architecture of host machine
+	GlobalService         string            `json:"global_service"`          // Global service string. If not-nil should be same as Service. (#614)
+}
+
+// checkEndpoint tries to connect to endpoint, which should be the full URL
+// returned by transport.endpoint. If the endpoint is not reachable,
+// checkEndpoint returns an error explaining why.
+func checkEndpoint(endpoint string) error {
+	req, err := http.NewRequest("POST", endpoint, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create http request: %v", err)
+	}
+	_, err = defaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// logStartup generates a startupInfo for a tracer and writes it to the log in
+// JSON format.
+func logStartup(t *tracer) {
+	tags := make(map[string]string)
+	for k, v := range t.globalTags {
+		tags[k] = fmt.Sprintf("%v", v)
+	}
+
+	info := startupInfo{
+		Date:                  time.Now().Format(time.RFC3339),
+		OSName:                osName(),
+		OSVersion:             osVersion(),
+		Version:               version.Tag,
+		Lang:                  "Go",
+		LangVersion:           runtime.Version(),
+		Env:                   t.config.env,
+		Service:               t.config.serviceName,
+		AgentURL:              t.transport.endpoint(),
+		Debug:                 t.config.debug,
+		AnalyticsEnabled:      !math.IsNaN(globalconfig.AnalyticsRate()),
+		SampleRate:            fmt.Sprintf("%f", t.rulesSampling.globalRate),
+		SamplingRules:         t.rulesSampling.rules,
+		Tags:                  tags,
+		RuntimeMetricsEnabled: t.config.runtimeMetrics,
+		HealthMetricsEnabled:  t.config.runtimeMetrics,
+		ApplicationVersion:    t.config.version,
+		Architecture:          runtime.GOARCH,
+		GlobalService:         globalconfig.ServiceName(),
+	}
+	if _, err := samplingRulesFromEnv(); err != nil {
+		info.SamplingRulesError = fmt.Sprintf("%s", err)
+	}
+	if err := checkEndpoint(t.transport.endpoint()); err != nil {
+		info.AgentError = fmt.Sprintf("%s", err)
+		log.Warn("DATADOG TRACER DIAGNOSTICS Unable to reach agent: %s", err)
```

Comment:
These log lines now conform to the specification and start with `Datadog Tracer v1.26.0 WARN: DATADOG TRACER DIAGNOSTICS`, but that seems like a lot of boilerplate.

Reply:
Why do we need the caps? This is horrible. It says "Datadog Tracer" twice on the same log line. @labbati can we change this?

Reply:
- `DATADOG TRACER CONFIGURATION`: I'd prefer to encourage solutions engs from our support team to always ask for the same thing: "Please search for 'DATADOG TRACER CONFIGURATION' and provide us that log line.".
- `DATADOG TRACER DIAGNOSTICS`: You can skip the 'DATADOG TRACER' part as this is meant for users and not to be transferred to our support team.